network/src/protocols/identity.rs

```diff
@@ -74,13 +75,23 @@ mod tests {
             .iter()
             .into(),
         );
-        let mut client_handshake = HandshakeMsg::new(chain_id, network_id);
-        client_handshake.add(
+        let server_handshake = HandshakeMsg {
+            chain_id,
+            network_id: network_id.clone(),
+            supported_protocols,
+        };
+        let mut supported_protocols = BTreeMap::new();
+        supported_protocols.insert(
```

Comment:
So, this changes the paradigm of adding supported protocols.  I didn't like the name `add` for the previous function, but I think `add_protocol` might be more apt.

The issue here with this change in paradigm, is now supported protocols need to be added ahead of time.

I do like having to put them in ahead of time, but what does this buy to be changed right now?

Reply:
This is a non-issue for now as this is what we do, so it's simpler. We can potentially add a method on UpgradeContext but I don't see the use, I also don't think HandshakeMsg should be part of UpgradeContext in its current form (if we want to optimize we need to cache the serialized HandshakeMsg instead).